Movie Application with Forms - Anca Nistor

915-Nistor-Anca opened the pull request.

Commit:
SQL Assignment

Commit:
sql & c# tasks

Commit:
sql code

915-Nistor-Anca committed:
Update EmployeeSalary.sql

eliminated getdate()

915-Nistor-Anca committed:
Update BankTransactions.sql

added "on delete cascade"

915-Nistor-Anca committed:
Update EmployeeSalary.sql

added "on delete cascade"

Commit:
added option to be able to print the salary of a given employee

Commit:
Movie Console Application done

915-Nistor-Anca committed:
update branch

915-Nistor-Anca committed:
update branch

915-Nistor-Anca committed:
update branch

915-Nistor-Anca committed:
update branch

915-Nistor-Anca committed:
update branch

915-Nistor-Anca committed:
movie app update

915-Nistor-Anca committed:
Delete MovieApplication/MovieService.cs

915-Nistor-Anca committed:
movie application with form

915-Nistor-Anca committed:
Merge branch 'MovieApplication_AncaNistor' of https://github.com/ZorB77/internship into MovieApplication_AncaNistor

915-Nistor-Anca committed:
movie application with form

915-Nistor-Anca committed:
update movie app with form

915-Nistor-Anca committed:
update app

915-Nistor-Anca committed:
database modifications created

added 2 more fields for tables Person and Role, deleted email field from Person

915-Nistor-Anca committed:
updates

915-Nistor-Anca committed:
updates

915-Nistor-Anca committed:
updates

ZorB77 commented on MovieApplication/MovieApplication/Models/Movie.cs:
```diff
@@ -0,0 +1,15 @@
+
+public class Movie
+{
+    public int movieID { get; set; }
```
the attributes of each class should start with capital letter

ZorB77 replied:
ID should be enough here, its not needed to specify again the name of the entity, we do this only when we use reference to another entity, like StudioID

ZorB77 commented on MovieApplication/MovieApplication/Services/MovieService.cs:
```diff
@@ -0,0 +1,205 @@
+    {
+        private MyDBContext db;
+        public MovieService(MyDBContext db) { this.db = db; }
+        public int addMovie(string name, DateTime releaseDate, string description, string genre)
```
make this method void

ZorB77 commented on MovieApplication/MovieApplication/Services/MovieService.cs:
```diff
@@ -0,0 +1,205 @@
+{
+    public class MovieService
+    {
+        private MyDBContext db;
```
add some empty line before the properties of the class and the constructor

ZorB77 commented on MovieApplication/MovieApplication/Services/MovieService.cs:
```diff
@@ -0,0 +1,205 @@
+
+        public string getMovieInformation(int movieID)
+        {
+            StringBuilder s = new StringBuilder();
```
use var

ZorB77 commented on MovieApplication/MovieApplication/Services/ReviewService.cs:
```diff
@@ -0,0 +1,197 @@
+            if (reviews.Any())
+            {
+                int averageRating = (int)reviews.Average(r => r.rating);
+                return averageRating;
```
if you have a return inside of the if, the else is not needed anymore

ZorB77 commented:
432 modified files 😿 you forgot to add something, but nice job!! 💯 for the windows form

915-Nistor-Anca committed:
web api application

created the controllers and the services; added annotations, swager; tested the api using postman.

915-Nistor-Anca committed:
Create VisualStudio.gitignore

915-Nistor-Anca committed:
Delete VisualStudio.gitignore

915-Nistor-Anca committed:
added logger class and model for mapping

915-Nistor-Anca committed:
Merge branch 'MovieApplication_AncaNistor' of https://github.com/ZorB77/internship into MovieApplication_AncaNistor
